### PR TITLE
Document IDE0240, 0241, 0250, 0251

### PR DESCRIPTION
**Pull request by gewarren**
Contributes to https://github.com/dotnet/docs/issues/30361.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/fundamentals/code-analysis/style-rules/ide0240.md](https://github.com/dotnet/docs/blob/e2474159b42ca02aeeec61ea467a6fafda2b00bf/docs/fundamentals/code-analysis/style-rules/ide0240.md) | [docs/fundamentals/code-analysis/style-rules/ide0240](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0240?branch=pr-en-us-36378) |
| [docs/fundamentals/code-analysis/style-rules/ide0241.md](https://github.com/dotnet/docs/blob/e2474159b42ca02aeeec61ea467a6fafda2b00bf/docs/fundamentals/code-analysis/style-rules/ide0241.md) | [docs/fundamentals/code-analysis/style-rules/ide0241](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0241?branch=pr-en-us-36378) |
| [docs/fundamentals/code-analysis/style-rules/ide0250.md](https://github.com/dotnet/docs/blob/e2474159b42ca02aeeec61ea467a6fafda2b00bf/docs/fundamentals/code-analysis/style-rules/ide0250.md) | [docs/fundamentals/code-analysis/style-rules/ide0250](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0250?branch=pr-en-us-36378) |
| [docs/fundamentals/code-analysis/style-rules/ide0251.md](https://github.com/dotnet/docs/blob/e2474159b42ca02aeeec61ea467a6fafda2b00bf/docs/fundamentals/code-analysis/style-rules/ide0251.md) | [docs/fundamentals/code-analysis/style-rules/ide0251](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0251?branch=pr-en-us-36378) |
| [docs/fundamentals/code-analysis/style-rules/index.md](https://github.com/dotnet/docs/blob/e2474159b42ca02aeeec61ea467a6fafda2b00bf/docs/fundamentals/code-analysis/style-rules/index.md) | [Code-style rules](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/index?branch=pr-en-us-36378) |
| [docs/fundamentals/code-analysis/style-rules/language-rules.md](https://github.com/dotnet/docs/blob/e2474159b42ca02aeeec61ea467a6fafda2b00bf/docs/fundamentals/code-analysis/style-rules/language-rules.md) | [Language and unnecessary rules](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/language-rules?branch=pr-en-us-36378) |

### CODE REVIEW

File: docs/fundamentals/code-analysis/style-rules/ide0241.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+- IDE0241
+dev_langs:
+- CSharp
+---
+# Nullable directive is unnecessary (IDE0241)
+
+| Property                 | Value                                                 |
+|--------------------------|-------------------------------------------------------|
+| **Rule ID**              | IDE0241                                               |
+| **Title**                | Nullable directive is unnecessary                     |
```
Comment (mavasani) [inline docs/fundamentals/code-analysis/style-rules/ide0241.md:18]: Tagging @sharwell to review documentation for IDE0240 and IDE0241

**Review by mavasani — APPROVED** [review]
Overall, LGTM

**Review by BillWagner — APPROVED** [review]
This also LGTM @gewarren 

I'm ready to :shipit:

File: docs/fundamentals/code-analysis/style-rules/ide0241.md (hunk 1 above)
Comment (gewarren) [reply docs/fundamentals/code-analysis/style-rules/ide0241.md:18]: I'm going to merge this now so I can start on the next batch of rule without conflicts, but feel free to leave feedback here and I'll address it in a separate PR.

**Pull request merged**